pan/strings.pan

```diff
@@ -0,0 +1,34 @@
+@documentation{
+Convenience string types intended for use in building up more complex types.
+}
+declaration template pan/strings;
+
+
+@documentation{
+A string that must contain at least one character (anything other than an empty string).
+}
+type string_non_empty = string(1..);
+
+
+@documentation{
+A string that must contain at least one character and may not contain any whitespace.
+}
+type string_non_whitespace = string_non_empty with !match(SELF, '\s');
+
+
+@documentation{
+A string that must contain at least one character, may contain whitespace, but may not start or end with whitespace.
+}
+type string_trimmed = string_non_empty with match(SELF, '^\S(.*\S)?$');
```

Comment:
I am not sure about this name. It's important that something that appears in a file far away from where it's defined is obvious to the reader. Does trimming something always mean whitespace? A quick google shows excel and jquery functions that do so maybe it is obvious.

Reply:
I'm keen to make sure these names make sense, so very happy to take suggestions.

AFAIK most languages call the function that does this either `trim` or `strip` and `trim` seems to be slightly more common.

Reply:
FWIW a quick survey gave me:

Language | Function Name
--|--
.NET | `Trim()`
Arduino | `trim()`
Excel | `trim()`
Go | `Trim()`
Java | `trim()`
JS | `trim()`
PHP | `trim()`
Python  | `strip()`
R | `trimws()`
Ruby | `strip()`